kong/dao/cassandra/base_dao.lua

```diff
@@ -152,11 +152,18 @@ function BaseDao:_open_session()
     return nil, DaoError(err, error_types.DATABASE)
   end
 
-  ok, err = session:set_keyspace(self._properties.keyspace)
-  if not ok then
+  local times, err = session:get_reused_times()
+  if err and err ~= "luasocket does not support reusable sockets" then
     return nil, DaoError(err, error_types.DATABASE)
   end
 
+  if times == 0 or not times then
```

Comment:
Oops my bad, it was a still needed check. About the order, it is correct. Most of the time while in use, and in need of performance (when Kong will be running), `get_reused_times()` will return a number (0 possibly). And thus our first condition need to be that.

If we are in a phase of NGINX that doesn't support the cosocket API (`init_by_lua`) or maybe not even in the NGINX context (the DAO is also used by the CLI or by the unit tests), `get_reused_times()` can also error and return nil. That is a less common case and not in need of performance.

Hence why the check. My bad for thinking I could remove it, because I reorganized the code before pushing but overlooked at it. I closed this PR by mistake, and had to reopen it in #197 tho